src/network_models/powerflow_evaluation.jl

```diff
@@ -0,0 +1,207 @@
+function add_power_flow_data!(::OptimizationContainer, ::Nothing, ::PSY.System)
+    # NO OP function
+    return
+end
+
+function _add_branches_aux_variables!(
+    container::OptimizationContainer,
+    vars::Vector{DataType},
+    branch_types::Vector{DataType},
+    branch_lookup::Dict{String, Int},
+)
+    branch_type_map =
+        Dict{String, DataType}(k => branch_types[v] for (k, v) in branch_lookup)
+    time_steps = get_time_steps(container)
+    for var_type in vars
+        for D in Set(branch_types)
+            branch_names = [k for (k, v) in branch_type_map if v == D]
+            add_aux_variable_container!(
+                container,
+                var_type(),
+                D,
+                branch_names,
+                time_steps,
+            )
+        end
+    end
+    return
+end
+
+function _add_buses_aux_variables!(
+    container::OptimizationContainer,
+    vars::Vector{DataType},
+    bus_lookup::Dict{Int, Int},
+)
+    time_steps = get_time_steps(container)
+    for var_type in vars
+        add_aux_variable_container!(
+            container,
+            var_type(),
+            PSY.ACBus,
+            sort!(collect(keys(bus_lookup))),
+            time_steps,
+        )
+    end
+    return
+end
+
+const ACTIVE_POWER_INJECTION_KEYS = [
+    ActivePowerVariable
+    PowerOutput
+    ActivePowerTimeSeriesParameter
+]
+
+function _make_injection_map!(container::OptimizationContainer, sys::PSY.System)
```

Comment:
@daniel-thom @GabrielKS This loop deserves son good review. This was the best approach I could come up with to find the index in the PoweFlowData matrices that correspond to the values that need to be read from the optimization container. 

What I am doing here is using a const for the types of keys that I want to capture, an additional alternative is to use traits and implement `is_injection_key(var)` but I went with this instead for simplicity. 

The function has comments about what is doing but if there are questions please let me know. I am not 100% confident on this implementation being the best.